cmd/legacy/daemon.go

```diff
@@ -277,6 +279,15 @@ func (d *Daemon) Start() error {
 	if err != nil {
 		mainLogger.Fatal("Failed to create controller manager", zap.Error(err))
 	}
+	// Create a channel to listen for OS signals.
+	sigChan := make(chan os.Signal, 1)
+	signal.Notify(sigChan, syscall.SIGINT, syscall.SIGTERM)
+	go func() {
+		sig := <-sigChan
+		mainLogger.Info("Received signal, stopping controller manager", zap.String("signal", sig.String()))
+		controllerMgr.Stop(ctx)
```

Comment:
I think it's cleaner to pass the context to (everything), and then cancel the context in the signal handler to cause cascading graceful shutdown.
controller-runtime also provides a [cross-platform helper which we could use](https://github.com/kubernetes-sigs/controller-runtime/blob/v0.6.5/pkg/manager/signals/signal.go#L29).

Reply:
ok nvm, apparently, we are using `SetupSignalHandler` and passing down the returned ctx already 🥲, and also TIL it is not a good idea to test with `kill -9`

Reply:
can't catch that one